src/vcl/Video.cc

```diff
@@ -201,12 +199,22 @@ void Video::read(const std::string &video_id)
     _operations.push_back(std::make_shared<Read>());
 }
 
+bool Video::is_read(void)
+{
+    return (_size.frame_count > 0);
+}
+
 void Video::perform_operations()
 {
     try
     {
-        for (int x = 0; x < _operations.size(); ++x) {
-            std::shared_ptr<Operation> op = _operations[x];
+        bool is_empty = _operations.empty();
+        auto first_op = !is_empty ? _operations.front() : NULL;
+        if (is_empty || (first_op && first_op->get_type() != READ)) {
```

Comment:
this can be simplified as (it is safe as .empty is evaluated first):
if ( (_operations.empty() || _operations.front()->get_type() != READ) && !is_read()) ) {
            if (!_video_id.empty())
                  throw
           _operations.push_front(std::make_shared<Read>());

Reply:
Yes, looks way more clean.